### PR TITLE
Wrap ref resolution errors with context

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

See https://github.com/crossplane/crossplane/issues/1499

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/database/v1alpha3/referencers.go [hunk 1]
```diff
@@ -139,7 +140,7 @@ func (mg *MySQLServerFirewallRule) ResolveReferences(ctx context.Context, c clie
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.resourceGroupName")
```
Comment (muvaf) [inline apis/database/v1alpha3/referencers.go:143]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.resourceGroupName")
```

File: apis/database/v1alpha3/referencers.go [hunk 2]
```diff
@@ -153,7 +154,7 @@ func (mg *MySQLServerFirewallRule) ResolveReferences(ctx context.Context, c clie
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.serverName")
```
Comment (muvaf) [inline apis/database/v1alpha3/referencers.go:157]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.serverName")
```

File: apis/database/v1alpha3/referencers.go [hunk 3]
```diff
@@ -174,7 +175,7 @@ func (mg *PostgreSQLServerFirewallRule) ResolveReferences(ctx context.Context, c
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.resourceGroupName")
```
Comment (muvaf) [inline apis/database/v1alpha3/referencers.go:178]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.resourceGroupName")
```

File: apis/database/v1alpha3/referencers.go [hunk 4]
```diff
@@ -188,7 +189,7 @@ func (mg *PostgreSQLServerFirewallRule) ResolveReferences(ctx context.Context, c
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.serverName")
```
Comment (muvaf) [inline apis/database/v1alpha3/referencers.go:192]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.serverName")
```

File: apis/database/v1alpha3/referencers.go [hunk 5]
```diff
@@ -209,7 +210,7 @@ func (mg *CosmosDBAccount) ResolveReferences(ctx context.Context, c client.Reade
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.resourceGroupName")
```
Comment (muvaf) [inline apis/database/v1alpha3/referencers.go:213]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.resourceGroupName")
```

File: apis/database/v1beta1/referencers.go [hunk 6]
```diff
@@ -39,7 +40,7 @@ func (mg *MySQLServer) ResolveReferences(ctx context.Context, c client.Reader) e
 		Extract:      reference.ExternalName(),
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, "spec.resourceGroupName")
```
Comment (muvaf) [inline apis/database/v1beta1/referencers.go:43]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.resourceGroupName")
```

File: apis/database/v1beta1/referencers.go [hunk 7]
```diff
@@ -60,7 +61,7 @@ func (mg *PostgreSQLServer) ResolveReferences(ctx context.Context, c client.Read
...
```
Comment (muvaf) [inline apis/database/v1beta1/referencers.go:64]:
```suggestion
		return errors.Wrap(err, "spec.forProvider.resourceGroupName")
```

File: apis/database/v1alpha3/referencers.go (hunk 1 above)
Comment (muvaf) [reply apis/database/v1alpha3/referencers.go:143]: These two suggestions do not seem to be addressed.

**Review by hasheddan — APPROVED** [review]
LGTM!

**Pull request merged**